modular_skyrat/modules/reagent_forging/code/anvil.dm

```diff
@@ -32,46 +32,62 @@
 
 /obj/structure/reagent_anvil/tong_act(mob/living/user, obj/item/tool)
 	var/obj/item/forging/forge_item = tool
-	var/obj/item/forging/incomplete/search_incomplete_src = locate(/obj/item/forging/incomplete) in contents
+	var/obj/obj_anvil_search = locate() in contents
 	if(forge_item.in_use)
 		to_chat(user, span_warning("You cannot do multiple things at the same time!"))
 		return FALSE
-	var/obj/item/forging/incomplete/search_incomplete_item = locate(/obj/item/forging/incomplete) in forge_item.contents
-	if(search_incomplete_src && !search_incomplete_item)
-		search_incomplete_src.forceMove(forge_item)
+	var/obj/obj_tong_search = locate() in forge_item.contents
+	if(obj_anvil_search && !obj_tong_search)
+		obj_anvil_search.forceMove(forge_item)
 		update_appearance()
 		forge_item.icon_state = "tong_full"
 		return FALSE
-	if(!search_incomplete_src && search_incomplete_item)
-		search_incomplete_item.forceMove(src)
+	if(!obj_anvil_search && obj_tong_search)
+		obj_tong_search.forceMove(src)
 		update_appearance()
 		forge_item.icon_state = "tong_empty"
 		return FALSE
 
 /obj/structure/reagent_anvil/hammer_act(mob/living/user, obj/item/tool)
+	//regardless, we will make a sound
 	playsound(src, 'modular_skyrat/modules/reagent_forging/sound/forge.ogg', 50, TRUE, ignore_walls = FALSE)
-	var/obj/item/forging/incomplete/search_incomplete_src = locate(/obj/item/forging/incomplete) in contents
-	if(!search_incomplete_src)
-		return FALSE
-	if(COOLDOWN_FINISHED(search_incomplete_src, heating_remainder))
-		to_chat(user, span_warning("You mess up, the metal was too cool!"))
+	//do we have an incomplete item to hammer out? if so, here is our block of code
+	var/obj/item/forging/incomplete/search_incomplete_src = locate() in contents
```

Comment:
I feel like that's a really weird name that loses all value past this line. It looks really weird to make manipulations on `search_incomplete_src`, it doesn't really tell me exactly what it's meant to be.